code/modules/projectiles/guns/ballistic.dm

```diff
@@ -138,7 +138,7 @@
 	return boolets
 
 /obj/item/weapon/gun/ballistic/suicide_act(mob/user)
-	if (chambered && chambered.BB && !chambered.BB.nodamage)
+	if (chambered && chambered.BB && can_trigger_gun(user) && !chambered.BB.nodamage)
```

Comment:
It jumps to line 152 where they pretend to shoot themselves and just fall over ded.